packages/flame_tiled/lib/src/renderable_tile_map.dart

```diff
@@ -23,12 +23,16 @@ class RenderableTiledMap {
   /// The size of tile to be rendered on the game.
   final Vector2 destTileSize;
 
+  /// Cached list of [SpriteBatch]es, ordered by layer.
+  final List<Map<String, SpriteBatch>>? batchesByLayer;
```

Comment:
Can't the `final Map<String, SpriteBatch> batches;` map be removed and only this map be populated instead?
Either the `batches` could be turned into a getter, or just removed completely, it should mostly be used internally right?

Reply:
Then we get rid of a lot of nullchecks too.

Reply:
I figured that would make this a breaking change, since RenderableTiledMap exposes batches in the public constructor, so I left it in for backwards compatibility. If you're confident that no one is using that constructor externally, then I can make the change. please let me know?

If we make the change, the getter would also be a dummy getter and just point to the first map in the list.

Reply:
> If we make the change, the getter would also be a dummy getter and just point to the first map in the list.

Sgtm!

Reply:
@spydon, sorry but before pushing the update, I want to confirm how much of a breaking change you're ok with me making:

**Option 1 [Breaking Change]** - Remove the field and remove the parameter entirely from the constructor. Replace it with the list as a required parameter (changes a public-facing contract). Add a ```@Deprecated``` batches getter that returns first in the list or an empty map.

**Option 2 [Backwards Compatible Contract]** - Remove the field, but keep the public facing contract the same with the optional, additional list parameter (batches is just an unused parameter passed in, not stored). Add a ```@Deprecated``` batches getter that returns first in the list or an empty map. I'll need to ignore the linter rule for unused parameters here.
It's worth noting that if anyone is directly using the RenderableTiledMap constructor, and passing in pre-loaded batches, their map would not render. This is still a breaking change from a functional perspective.

Let me know which is preferable and I will push. Thanks!

Reply:
Sorry for the slow reply, I had to double check with the rest of the team first. Since this is for a bridge library and not the core library you can go with option 1 (this library was accidentally put at version 1.0.0, it really should be <1 still).

Reply:
Thanks for checking, just updated to proceed with Option 1.